grails-app/services/com/unifina/service/MetricsService.groovy

```diff
@@ -0,0 +1,46 @@
+package com.unifina.service
+
+import com.unifina.data.EventQueueMetrics
+import grails.compiler.GrailsCompileStatic
+
+import javax.management.ObjectName
+import java.lang.management.ManagementFactory
+
+@GrailsCompileStatic
+class MetricsService {
+	SignalPathService signalPathService
+
+	int numOfRunningCanvases() {
+		signalPathService.runningSignalPaths.size()
+	}
+
+	int bytesUsedByRunningCanvases() {
+		return 0 // TODO
```

Comment:
How about this TODO?

Reply:
The problem is that we cannot serialize a canvas while it is running because it involves nulling fields etc. We could turn on the serialization of canvases during regular intervals, but that would affect their real-time performance guarantees.

Reply:
I suggest we leave this like this for now and work on it later so that we get at least some metrics online.

Reply:
Ok, please file JIRA ticket to follow up on this.

Are you worried about concurrency issues? The collection for this metric could happen in the canvas event queue (similarly to the normal serialization at regular intervals), yes?